sierra2mlir/src/compiler/mod.rs

```diff
@@ -44,7 +67,56 @@ impl<'ctx> Compiler<'ctx> {
         program: &'ctx Program,
         main_print: bool,
         print_fd: i32,
+        available_gas: Option<usize>,
     ) -> color_eyre::Result<Self> {
+        let sierra_program_registry = ProgramRegistry::<CoreType, CoreLibfunc>::new(program)?;
+
+        // Create the compiler gas structure if gas was provided, similar to cairo casm runner.
+        let gas: Option<CompilerGas> = if let Some(available_gas) = available_gas {
+            let function_set_costs: OrderedHashMap<FunctionId, OrderedHashMap<CostTokenType, i32>> =
+                Default::default();
+
+            let pre_function_set_costs = function_set_costs
```

Comment:
This whole block could probably do with some step by step comments as to what each of these maps is for and why it's collected the way that it is. It's a little difficult to process as is

Reply:
i mostly copied this from the cairo-runner crate